tests/HttpExtractorTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\HttpExtractor;
+
+use function file_get_contents;
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Psr7\Uri;
+use PHPUnit\Framework\TestCase;
+use function sys_get_temp_dir;
+
+class HttpExtractorTest extends TestCase
+{
+    /** @var mixed[] */
+    private $history = [];
+
+    public function testExtract(): void
+    {
+        $resource = new Uri('http://example.com/result.txt');
```

Comment:
Ideálně bych to viděl v něčem, co si ten test nebo developer sám nasetupuje. Můžeš pak snadno simulovat 404, 403, 500, https, potenciálně to můžeš snadno rozšiřovat o přihlašování apod. Může to bejt něco v AWS (statický fajly v S3) nebo to může bejt třeba nějaká docker servisa s jednoduchým HTTP serverem. Tady a teď to je asi overkill, ale už ta 404 na `example.com` nepůjde :-)

Reply:
Ale půjde, viz nový test pod tím. Ale v rámci integračního testu to bude potřeba. To jsme se už bavili. Ale s tím budu asi potřebovat nějak pomoct.

Reply:
Ježiš, jsem si nevšiml, že ten Guzzle je mockovanej, samozřejmě, omg! 404 a vše ostatní není problém. Integrační test naladíme, to nebude problém, co všechno tam budeš chtít testovat?